### PR TITLE
Fix typo that caused the ratings to stay

### LINKED ISSUE #2829: Ratings dialog not disappearing after inactivity
## 🐛 Bug Report
From the specs in https://github.com/toggl/mobileapp/issues/1972
> - When the app goes into background and user comes back before 5 mis has passed, he/she can still see the popup. After 5 min, the popup will be hidden.

This does not seem to happen. I've gone back to the app multiple times now, including after having it in the background for 20 minutes, and it's still there.

## Expected behavior
What the specs say:
> - When the app goes into background and user comes back before 5 mis has passed, he/she can still see the popup. After 5 min, the popup will be hidden.

I believe the intention was to snooze it for 24h - but I don't see that in the specs right now. Am I confusing something, or did I miss it? Or should it just happen on the next time the trigger happens?

Comment from @pe0ny would be appreciated as well.

## To reproduce
- Get the dialog to appear
- Don't interact with it, put app in background
- Come back after a while
- It's still there

### PR DESCRIPTION
**Pull request by dshr**
## What's this?
A fix to the clingy ratings view that just wouldn't go away. Turns out it was just a small type causing this.

### Relationships
Closes #2829

## Why do we want this?
To not have an ever-present rating view.

## How is it done?
By setting the `ToggleRatingViewVisibilityHint`'s `ForceHide` parameter to `true`

### Why not another way?
¯\_(ツ)_/¯

### Side effects
None.

## Review hints
1. Grab a timer
2. Get the rating view to show up and start the timer.
3. Move the app to background.
4. Open the app again.
5. Wait for the timer to run out and see the rating view disappear.
6. Set date to 24h later and repeat steps 1-5.
7. The rating view won't bother you anymore.

## :squid: Permissions
✔️ go for it!

### CODE REVIEW

File: Toggl.Foundation.MvvmCross/ViewModels/MainViewModel.cs [hunk 1]
```diff
@@ -304,7 +304,7 @@ private void presentRatingViewIfNeeded(bool shouldBevisible)
             onboardingStorage.SetRatingViewOutcome(RatingViewOutcome.NoInteraction, timeService.CurrentDateTime);
             timeService.RunAfterDelay(ratingViewTimeout, () =>
             {
-                navigationService.ChangePresentation(new ToggleRatingViewVisibilityHint());
+                navigationService.ChangePresentation(new ToggleRatingViewVisibilityHint(true));
```
[the remark this answers is not in the document]
Comment (dshr) [reply Toggl.Foundation.MvvmCross/ViewModels/MainViewModel.cs]: I'm not a huge fan of `ForceHide` as well, can we make it into `Visible` or something? I'm just not aware of the reasoning for the `Force-` part of it.

**Comment by amulware** [comment]
RE @simonrozsival Maybe it shouldn't be a `bool` either, but a nicely named parameter to make it clear what 'true' means :D (or have two static construction methods on the type instead of using the constructor, whatever is nicer).

File: Toggl.Foundation.MvvmCross/ViewModels/MainViewModel.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (simonrozsival) [reply Toggl.Foundation.MvvmCross/ViewModels/MainViewModel.cs]: Feel free to rename it, that is fine. Let's just not rewrite the whole showing/hiding mechanism in a bugfix PR for a hotfix release.

**Comment by dshr** [comment]
In `TogglPresenter.cs:205` we're receiving the `ToggleRatingViewVisibilityHint` and based on it's `ForceHide` param we either show or hide the ratings view.

The two static constructors sound like the clearest option here imo.

**Comment by simonrozsival** [comment]
@amulware I don't like the concept of presentation hints from the business logic layer.. this way of communication seems wrong to me.

I think we could change this presentation hint to an observable and use it instead to show/hide the rating view. But in a different PR ofc. I will add an issue.

**Comment by sofokli1** [comment]
My approval still counts

**Pull request merged**